routes/companies.js

```diff
@@ -0,0 +1,152 @@
+var express = require('express');
+var router = express.Router();
+var Company = require('../models/company')
+
+router.get('/', function (request, response, next) {
+  index(request, response)
+});
+
+router.get('/new', function (request, response, next) {
+  response.render('companies/new', { title: 'New Company', layout: './layouts/application' });
+});
+
+router.post('/', (request, response) => {
+  create(request, response)
+})
+
+router.get('/:id/edit', (request, response) => {
+  edit(request, response)
+})
+
+router.get('/:id', (request, response) => {
+  show(request, response)
+})
+
+router.post('/:id', (request, response) => {
+  update(request, response)
+})
+
+router.get('/:id/delete', (request, response) => {
+  destroy(request, response)
+})
+
+async function index(request, response) {
+  let db = request.app.get('db')
```

Comment:
It looks like you are doing this in each route, just to pass the `db` to the model to fetch the information.

Who is interested in having a connection to the database handy? The views? The controller? The router? The model?

Reply:
Yes it was going to be one of my questions, I would like to be able to access the same `db` object in the model, without re-instantiating it every time I call a `Model` function